securedrop_export/archive.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python3
+
+import datetime
+import json
+import logging
+import os
+import tempfile
+
+from securedrop_export.exceptions import ExportException
+from securedrop_export.status import BaseStatus
+from securedrop_export.command import Command
+from securedrop_export.directory_util import safe_extractall
+
+logger = logging.getLogger(__name__)
+
+
+class Status(BaseStatus):
+    ERROR_ARCHIVE_METADATA = "ERROR_ARCHIVE_METADATA"
+    ERROR_METADATA_PARSING = "ERROR_METADATA_PARSING"
+    ERROR_EXTRACTION = "ERROR_EXTRACTION"
+
+
+class Metadata(object):
+    """
+    Object to parse, validate and store json metadata from the sd-export archive.
+    """
+
+    METADATA_FILE = "metadata.json"
+    SUPPORTED_ENCRYPTION_METHODS = ["luks"]
+
+    def __init__(self, archive_path: str):
+        self.metadata_path = os.path.join(archive_path, self.METADATA_FILE)
+
+    def validate(self) -> "Metadata":
+        # Read metadata json and set relevant attributes
+        try:
+            with open(self.metadata_path) as f:
```

Comment:
if we want to resolve https://github.com/freedomofpress/securedrop-export/issues/9#issuecomment-531390692, we could store the passphrase in memory and delete it from the metadata.json file here. (And still delete the metada.json file at the end) @gonzalo-bulnes thoughts/feelings?

Reply:
Unless I'm misunderstanding something, the archive is already persisted in disk at this point, isn't it?

Of course we can remove the passphrase for the file, and let's not better be enemy of good, but the way I understand @creviera's comment is that it would be good for the passphrase not to be persited on disk in the first place. If we wanted to prevent the passphrase from being written on disk, my intuition would be to look into ways to not include it in the archive at any point. (I'm assuming that the archive will be written on disk as part of an underlying Qubes OS process that we probably don't want to modify, please correct me if I'm wrong).

Only for illustration: I think that (coincidentally) what's being discussed in #112 could be one way to do that. (The passphrase could be sent _alongside_ the archive, but not inside it. Unlike what's part of the Qubes OS-driven file transfer, we would have entire control over what happens with any additional parameters included with the request/command.) _I may have gotten a hammer, but this looks to me like it could be a nail._ But I'm not intensing narrow the exploration or even get into solution mode prematurely, this is just to clarify what @creviera's suggestion inspires me!

Reply:
yeah you're right about it already having been written to disk. (I was proposing deleting metadata.json at this point, but it would definitely be better to pass the password and never write it to the archive)